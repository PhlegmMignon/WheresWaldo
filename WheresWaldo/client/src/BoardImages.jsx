import convention from "./assets/nMpQXwq.jpg";
import convention_kon from "./assets/kon.webp";
import convention_luffy from "./assets/luffy.png";
import convention_zim from "./assets/Zim.yelling.svg";

import stage2 from "./assets/stage2.jpg";

export default function BoardImages() {
  const getImages = () => {
    return [
      {
        name: "convention",
        src: convention,
        id: 0,
        characters: [
          { name: "zim", src: convention_zim, id: 0 },
          { name: "luffy", src: convention_luffy, id: 1 },
          { name: "kon", src: convention_kon, id: 2 },
        ],
        style: ["max-content", "max-content"],
      },
      {
        name: "stage 2",
        src: stage2,
        id: 1,
        characters: [
          { name: "zim", src: convention_zim, id: 0 },
          { name: "luffy", src: convention_luffy, id: 1 },
          { name: "kon", src: convention_kon, id: 2 },
        ],
        style: ["100%", "max-content"],
      },
    ];
  };

  return { getImages };
}
