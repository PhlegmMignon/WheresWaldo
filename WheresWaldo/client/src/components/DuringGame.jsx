import { useState, useEffect } from "react";
import Dropdown from "./Dropdown";

export default function DuringGame({
  gameImage,
  found,
  setFoundStatus,
  setGameState,
  updateTimer,
}) {
  const [coordinate, setCoordinate] = useState([0, 0]);
  const [dropdown, toggleDropdown] = useState(false);

  //Dropdown controls
  const handleClick = (e, slider) => {
    // e.preventDefault();
    let x = e.pageX + slider.scrollLeft;
    let y = e.pageY + slider.scrollTop;
    // console.log(x + " " + y);
    setCoordinate([x, y]);
  };

  // const assignPosition = () => {};

  useEffect(() => {
    console.log(gameImage);

    //Makes image draggable
    let isDragged = false;
    let mouseDown = false;
    let startX, scrollLeft, startY, scrollTop;
    const slider = document.getElementById("imageContainer");

    const move = (e) => {
      e.preventDefault();

      if (!mouseDown) return;
      isDragged = true;
      // console.log("move");
      const x = e.pageX - slider.offsetLeft;
      const scroll = x - startX;
      slider.scrollLeft = scrollLeft - scroll;

      const y = e.pageY - slider.offsetTop;
      const Yscroll = y - startY;
      slider.scrollTop = scrollTop - Yscroll;
    };

    const startDragging = (e) => {
      isDragged = false;
      // console.log("dragging");
      mouseDown = true;

      startX = e.pageX - slider.offsetLeft;
      scrollLeft = slider.scrollLeft;

      startY = e.pageY - slider.offsetTop;
      scrollTop = slider.scrollTop;
    };

    const stopDragging = () => {
      // console.log("stopDrag");
      mouseDown = false;
    };

    slider.addEventListener("mousemove", move, false);
    slider.addEventListener("mousedown", startDragging, false);
    slider.addEventListener("mouseup", stopDragging, false);
    slider.addEventListener("mouseleave", stopDragging, false);

    //Dropdown controls
    let ele = document.getElementById("imageContainer");
    ele.addEventListener("mouseup", (e) => {
      handleClick(e);
      dropdown ? toggleDropdown(false) : toggleDropdown(true);
    });

    return () => {
      let newSlider = slider.cloneNode(true);
      slider.parentNode.replaceChild(newSlider, slider);
    };
  }, [gameImage]);

  setInterval(() => {
    updateTimer();
  }, [1000]);

  return (
    <div id="imageContainer" className="">
      <img id="image" src={gameImage.src} alt="" className="" />
      <Dropdown />
    </div>
  );
}

//click - dragFalse - openModal -
