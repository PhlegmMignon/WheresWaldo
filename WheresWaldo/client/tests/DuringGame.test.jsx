import { render } from "@testing-library/react";
import { vi } from "vitest";
import DuringGame from "../src/components/DuringGame.jsx";
import BoardImages from "../src/BoardImages.jsx";

describe("DuringGame unit tests", () => {
  it("Renders DuringGame correctly", async () => {
    const gameImage = BoardImages().getImages();
    const fakeGetScores = vi.fn();

    const { container } = render(
      <DuringGame
        gameImage={gameImage[0]}
        modalFound={false}
        getScores={fakeGetScores}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
