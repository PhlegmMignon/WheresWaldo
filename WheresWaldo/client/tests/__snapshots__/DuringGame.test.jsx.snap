// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DuringGame unit tests > Renders DuringGame correctly 1`] = `
<div>
  <div
    id="imageContainer"
  >
    <img
      alt="picture of convention"
      id="image"
      src="/src/assets/nMpQXwq.jpg"
    />
  </div>
</div>
`;

exports[`DuringGame unit tests > Renders stage select 1`] = `
<div>
  <div
    id="imageContainer"
  >
    <img
      alt="picture of convention"
      id="image"
      src="/src/assets/nMpQXwq.jpg"
    />
  </div>
</div>
`;
